ethcore/sync/src/chain/handler.rs

```diff
@@ -654,9 +672,13 @@ impl SyncHandler {
 			trace!(target: "sync", "{} Ignoring packet from unconfirmed/unknown peer", peer_id);
 			return Ok(());
 		}
-
+		if sync.private_tx_handler.is_none() {
```

Comment:
why not use pattern matching/if let to eliminate those `expect()` i.e, something like

```rust
let private_handler = match sync.private_tx_handler {
  Some(p) => p,
  None => return Ok(()), 
}
```